enhancements/cluster-logging/workload-identity-auth-for-azure-monitor-logs.md

```diff
@@ -0,0 +1,295 @@
+  #[configurable(metadata(docs::examples = "qWgdYAmab0YSkuL1qKv5bPX"))]
+  pub client_secret: Option<String>,
+
+  /// The [Federated Token File Path][federated_token_file_path] pointing to a federated token for authentication.
+  ///
+  /// [token_path]: https://learn.microsoft.com/en-us/entra/identity-platform/v2-oauth2-client-creds-grant-flow#third-case-access-token-request-with-a-federated-credential
+  #[configurable(metadata(docs::examples = "/path/to/my/token"))]
+  pub federated_token_file_path: Option<String>,
+
+  /// The [Tenant ID][tenant_id] for authentication.
+  ///
+  /// The directory tenant the application plans to operate against, in GUID or domain-name format.
+  ///
+  /// [tenant_id]: https://learn.microsoft.com/en-us/azure/azure-monitor/data-collection/data-collection-rule-structure#properties
+  #[configurable(metadata(docs::examples = "11111111-2222-3333-4444-555555555555"))]
+  pub tenant_id: String,
+
+  /// The [Client ID][client_id] for authentication.
+  ///
+  /// The client ID is the application ID that's assigned to your app. You can find this information in the portal where you registered your app.
+  ///
+  /// [client_id]: https://learn.microsoft.com/en-us/entra/identity-platform/v2-oauth2-client-creds-grant-flow#third-case-access-token-request-with-a-federated-credential
+  #[configurable(metadata(docs::examples = "11111111-2222-3333-4444-555555555555"))]
+  pub client_id: String,
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+1. Relies on [this upstream vector PR](https://github.com/vectordotdev/vector/pull/22912) to implement the Azure Log Ingestion sink utilizing the Log Ingestion API. This is a separate sink from the data collector API. We can support both APIs while transitioning.
+2. Current `master` branch of [upstream Vector](https://github.com/vectordotdev/vector), `>=v0.46.1`, as of 04/29/2025, utilizes `azure_identity@v0.21.0` which relies solely on environment variables for workload identity credentials and will not be sufficient when forwarding logs to multiple different Azure sinks. The aforementioned PR relies on `azure_identity@v0.21.0`.
+    - `azure_identity@v0.23.0` allows for setting `client_id`, `tenant_id`, etc. for authentication.
+    - [Workload Identity Credentials azure_identity@v0.23.0 SDK Ref](https://github.com/Azure/azure-sdk-for-rust/blob/azure_identity%400.23.0/sdk/identity/azure_identity/src/credentials/workload_identity_credentials.rs)
+    - `v0.23.0` can also utilize the [ChainedTokenCredential](https://github.com/Azure/azure-sdk-for-rust/blob/azure_identity%400.23.0/sdk/identity/azure_identity/src/chained_token_credential.rs) struct which provides a user-configurable `TokenCredential` authentication flow for applications.
+3. Additional fields will be required in sink configuration in upstream Vector's API. See [proposed API](#proposed-api) above.
```

Comment:
Have you or considered mentioning the dependency and additional config in the upstream to see if it can be added as part of the initial impl? it may be worth trying

Reply:
I can ask about it in the upstream to see if they can update it.